### PR TITLE
Fix/improve edit address widget

### PR DESCRIPTION
**Pull request by StanislavPrime**
- [x] Update the value inside the edit address widget after the new address value is saved.
- [x] Remove the success message after the address is saved.
- [x] Move the X button in the modal sheet up so there is not so much space between the save button and the X button and there X button. does not overlaps with the notch of devices like iPhone, add safe area around the edit address widget.
- [x] Check that the change from above does not breaks any other widget content of another library.

### CODE REVIEW

File: lib/src/lib_edit_address/ui_components/edit_address_widget.dart [hunk 1]
```diff
@@ -178,40 +192,42 @@ class EditAddressWidget<T extends PickerItemModel> extends StatelessWidget {
...
+                  context.getEditAddressLocalizedStrings
+                      .permanentAddressContentMessage,
+              configuration: widget.configuration,
             );
       case UserProfileCardTypes.mailingAddress:
       case UserProfileCardTypes.email:
       case UserProfileCardTypes.phone:
         return () {
-          showAppAddressForCorrespondence<T>(
+          showEditAddressBottomSheet<T>(
```
Comment (StanevPrime) [inline lib/src/lib_edit_address/ui_components/edit_address_widget.dart:207]: See the comment above

**Review by StanevPrime — CHANGES_REQUESTED** [review]
In general looks better than before. Please see the comments as well as add golden tests for the not-empty cases

File: lib/src/lib_edit_address/ui_components/edit_address_form.dart [hunk 2]
```diff
@@ -116,7 +114,7 @@ class EditAddressForm<T extends PickerItemModel> extends StatelessWidget {
                 configuration: const TextFieldConfiguration(
                   haveOnlyOneSheet: false,
                 ),
-                dialogHasBottomPadding: false,
+                dialogHasBottomPadding: true,
```
Comment (RomanovaPrime) [inline lib/src/lib_edit_address/ui_components/edit_address_form.dart:117]: This will broke the storybook example. Alternative way to make it work is to expose the property, so we can easily fix the project.

File: test/lib_edit_address/factory/edit_address_widget.dart [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import '../service/search_country_service_mock.dart';
+import '../utils/edit_address_error_mapper_util.dart';
+
+/// Change the parameters according the the needs of the test
+Widget editAddressWidget({
+  AddressModel? onAddressSet,
+  bool? isLoading,
+  ErrorModel? errors,
+  String? street,
+  String? city,
```
Comment (RomanovaPrime) [inline test/lib_edit_address/factory/edit_address_widget.dart:20]: Name widgets with capital-camel-case and add Factory sufix

Comment (StanislavPrime) [reply test/lib_edit_address/factory/edit_address_widget.dart:20]: I renamed the method to createEditAddressWidget and moved it to edit_address_page_factory.dart file

File: lib/src/lib_edit_address/ui_components/edit_address_form.dart (hunk 2 above)
Comment (StanislavPrime) [reply lib/src/lib_edit_address/ui_components/edit_address_form.dart:117]: I have exposed the property for now from this file [edit_address_form.dart]

Comment (StanevPrime) [reply lib/src/lib_edit_address/ui_components/edit_address_form.dart]: As we decided the storybook won't be supported anymore

File: lib/src/lib_edit_address/ui_components/edit_address_widget.dart [hunk 4]
```diff
@@ -104,6 +104,18 @@ class EditAddressWidget<T extends PickerItemModel> extends StatelessWidget {
...
   );
 
+  @override
+  State<EditAddressWidget> createState() => _EditAddressWidgetState();
+}
+
+class _EditAddressWidgetState<T extends PickerItemModel>
+    extends State<EditAddressWidget<T>> {
+  AddressModel? savedModel;
+
```
[the remark this answers is not in the document]
Comment (StanislavPrime) [reply lib/src/lib_edit_address/ui_components/edit_address_widget.dart:114]: I have added the initialization in an `initState `

File: lib/src/lib_edit_address/ui_components/edit_address_form.dart (hunk 2 above)
Comment (StanislavPrime) [reply lib/src/lib_edit_address/ui_components/edit_address_form.dart]: I have removed hardcoded ` dialogHasBottomPadding: true,` in both `TextFieldDialog`

File: lib/src/lib_edit_address/ui_components/edit_address_bottom_sheet.dart [hunk 5]
```diff
@@ -7,8 +7,9 @@ import '../../../search_picker.dart';
 import '../../../widget_toolkit.dart';
 import '../views/edit_address_page.dart';
 
-void showAppAddressForCorrespondence<T extends PickerItemModel>(
+void showEditAddressBottomSheet<T extends PickerItemModel>(
   BuildContext context, {
+  required OnAddressSaved onAddressSaved,
```
[the remark this answers is not in the document]
Comment (StanislavPrime) [reply lib/src/lib_edit_address/ui_components/edit_address_bottom_sheet.dart]: This change was applied

File: lib/src/lib_edit_address/views/edit_address_page.dart [hunk 6]
```diff
@@ -114,89 +121,101 @@ class EditAddressPage<T extends PickerItemModel> extends StatelessWidget {
...
-                    MessagePanelWidget(
-                      message:
-                          editAddressLocalizedStrings?.addressChangedMessage ??
-                              context.getEditAddressLocalizedStrings
-                                  .addressChangedMessage,
-                      messageState: MessagePanelState.positiveCheck,
+              builder: (context, snapshot, bloc) {
+                if (snapshot.hasData) {
+                  SchedulerBinding.instance.scheduleFrameCallback((timeStamp) {
+                    onAddressSaved(snapshot.data!);
```
[the remark this answers is not in the document]
Comment (StanislavPrime) [reply lib/src/lib_edit_address/views/edit_address_page.dart]: I have replaced the listening for the new value to happen in a `RxBlocListener `

File: lib/src/lib_edit_address/ui_components/edit_address_widget.dart (hunk 1 above)
Comment (StanislavPrime) [reply lib/src/lib_edit_address/ui_components/edit_address_widget.dart]: This change was applied

File: lib/src/lib_ui_components/show_modal.dart [hunk 7]
```diff
@@ -240,10 +240,10 @@ class _ModalContent extends StatelessWidget {
       );
 
   Widget _buildCloseButton(BuildContext context) => Container(
-        padding: const EdgeInsets.symmetric(vertical: 10),
```
Comment (StanevPrime) [inline lib/src/lib_ui_components/show_modal.dart:243]: Please use the theme extension, instead of hardcoding values

**Review by StanevPrime — APPROVED** [review]
In general looks good. I left just one cosmetic comment

File: lib/src/lib_ui_components/show_modal.dart (hunk 7 above)
Comment (StanislavPrime) [reply lib/src/lib_ui_components/show_modal.dart:243]: This padding was added to theme extensions
